File: Python/refer-offer/cong_shang_dao_xia_da_yin_er_cha_shu_lcof.py
from typing import *

class TreeNode:
    def __init__(self, x):
        self.val = x
        self.left = None
        self.right = None

def levelOrder(root: TreeNode) -> List[int]:
    queue = Deque()
    queue.append(root)
    res = list()
    while len(queue) != 0:
        node = queue.popleft()
        if not node: continue
        if node.left: queue.append(node.left)
        if node.right: queue.append(node.right)
        res.append(node.val)
    
    return res

File: Java/src/com/dbc/algorithm/ReferOffer/CongShangDaoXiaDaYinErChaShuLcof.java
package com.dbc.algorithm.ReferOffer;

import java.util.*;

public class CongShangDaoXiaDaYinErChaShuLcof {
    public class TreeNode {
        int val;
        TreeNode left;
        TreeNode right;

        TreeNode(int x) {
            val = x;
        }
    }

    public int[] levelOrder(TreeNode root) {
        Queue<TreeNode> queue = new LinkedList<>();
        queue.offer(root);
        List<Integer> res = new ArrayList<>();
        while (!queue.isEmpty()) {
            TreeNode node = queue.poll();
            if (node == null) continue;
            if (node.left != null) queue.offer(node.left);
            if (node.right != null) queue.offer(node.right);
            res.add(node.val);
        }
        int[] result = new int[res.size()];
        for (int i = 0; i < res.size(); i++) {
            result[i] = res.get(i);
        }
        return result;
    }
}
